lib/matplotlib/axes/_axes.py

```diff
@@ -3194,9 +3198,16 @@ def errorbar(self, x, y, yerr=None, xerr=None,
         kwargs = {k: v for k, v in kwargs.items() if v is not None}
         kwargs.setdefault('zorder', 2)
 
-        if errorevery < 1:
-            raise ValueError(
-                'errorevery has to be a strictly positive integer')
+        try:
+            offset, errorevery = errorevery
+        except TypeError:
+            offset = 0
+
+        int_msg = 'errorevery must be positive integer or tuple of integers'
```

Comment:
It's more readable to define the message directly in the context of the `ValueError`. Also, this allows to make the error message more specific.

~~~
        if errorevery < 1 or int(errorevery) != errorevery:
            raise ValueError(
                'errorevery must be positive integer or tuple of integers')
        if int(offset) != offset:
            raise ValueError('errorevery offset must be an integer >= 0')
~~~